### ISSUE TITLE
Bazel dependencies issues

### ISSUE THREAD

**Issue by cpsauer** [google/google-toolbox-for-mac#410]
Hi wonderful Googlers,

I was debugging something else and noticed some potential problems in the bazel dependencies https://github.com/google/google-toolbox-for-mac/blob/main/bazel_support/repositories.bzl

1. The rules_cc version in there is super old--old enough that it can break some other things that depend on it. I think the solution, though, is probably to just eliminate it rather than update, since it looked like it was only used for objc_library, and my understanding is that Bazel had fallen back to just having people use the one that was automatically loaded rather than the one from rules_cc.
2. rules_apple needs to load a bunch of transitive dependencies and is also *very* old. (See release snippet [here](https://github.com/bazelbuild/rules_apple/releases/tag/3.1.1)). You could   have users call a bunch of layers of dependency macros...but it might be easier to just scrap all this dependency pulling and just tell people to get rules_apple. If they're using this, they're probably already pulling it in. 

TL;DR recommend emptying google_toolbox_for_mac_rules_dependencies, since the contents are super out of date and probably not worth it. (But keep the empty function around in case you need it in the future--and to avoid breaking existing users.)

Thoughts?

Thanks for all you do,
Chris
(ex-Googler)

**Comment by thomasvl** [comment]
The bazel support was externally contributed, so feel free to send a PR to update things.

**Comment by cpsauer** [comment]
Done. Here's what I'd advise #411
